thoth-tech/doubtfire-deploy
# feat(dashboard): display target grade change history

What does this PR do?

This PR adds a new feature to the progress dashboard which allows the users to view the history of target grade changes. It enhances transparency by displaying all past modifications to a student’s target grade.

Key Changes:

	•	Updated progress-dashboard.coffee:
	•	Implemented API calls to fetch targetGradeHistory.
	•	Processed the retrieved data for display on the dashboard.
	•	Modified progress-dashboard.tpl.html:
	•	Added a new section to display the target grade change history in a table format.
	•	Included columns for previous grade, new grade, changed by, and timestamp.

Why are these changes needed?

	•	Enhances User Experience:
	•	Provides students and instructors with a clear record of all target grade adjustments.
	•	Facilitates better communication and understanding between students and instructors regarding progress expectations.
	•	Improves Transparency and Accountability:
	•	Allows users to track changes over time, ensuring that any modifications are documented and accessible.

How should this be tested?

	1.	Prerequisites:
	•	Ensure the back-end API is updated with the TargetGradeHistory model and corresponding endpoints.
	•	Pull the latest changes from the feature/target-grade-history branch in the doubtfire-api repository.
	2.	Testing Steps:
	•	Log in as a student and navigate to the progress dashboard.
	•	Change the target grade multiple times to generate history entries.
	•	Verify that the history table updates accordingly, displaying all changes.
	•	Check that the information displayed (previous grade, new grade, changed by, timestamp) is accurate.
	•	Repeat the steps while logged in as an instructor to ensure consistency.

<img width="977" alt="Screenshot 2024-11-14 at 7 35 26 pm" src="https://github.com/user-attachments/assets/d99fa724-2ba6-4ece-a9ed-2ffb342343aa">
